packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/encounter-list.component.tsx

```diff
@@ -15,39 +15,52 @@ import {
   Button,
   OverflowMenu,
   OverflowMenuItem,
+  TableToolbar,
+  TableToolbarContent,
+  TableToolbarSearch,
+  DataTableHeader,
+  Dropdown,
 } from 'carbon-components-react';
 import Edit16 from '@carbon/icons-react/es/edit/16';
-import { useTranslation } from 'react-i18next';
-import {
-  formatDate,
-  formatDatetime,
-  formatTime,
-  parseDate,
-  useLayoutType,
-  usePagination,
-  usePatient,
-} from '@openmrs/esm-framework';
+import { formatDatetime, formatTime, parseDate, useLayoutType, usePagination } from '@openmrs/esm-framework';
 import { formEntrySub, launchPatientWorkspace, PatientChartPagination } from '@openmrs/esm-patient-common-lib';
-import { FormattedEncounter } from '../visit-detail-overview.component';
-import styles from '../visit-detail-overview.scss';
-import { format } from 'prettier';
+import { MappedEncounter } from './visit-summary.component';
+import EncounterObservations from './encounter-observations.component';
+import styles from './encounter-list.scss';
 
 interface EncounterListProps {
-  encounters: Array<FormattedEncounter>;
-  isShowingAllEncounters?: boolean;
+  encounters: Array<MappedEncounter>;
+  showAllEncounters?: boolean;
 }
 
-const EncounterList: React.FC<EncounterListProps> = ({ isShowingAllEncounters, encounters }) => {
+type FilterProps = {
+  rowIds: Array<string>;
+  headers: Array<DataTableHeader>;
+  cellsById: any;
+  inputValue: string;
+  getCellId: (row, key) => string;
+};
+
+const EncounterList: React.FC<EncounterListProps> = ({ showAllEncounters, encounters }) => {
   const encountersCount = 20;
   const { t } = useTranslation();
-  const { patient } = usePatient();
-  const isTablet = useLayoutType() === 'tablet';
+  const encounterTypes = [...new Set(encounters.map((encounter) => encounter.encounterType))];
   const { results: paginatedEncounters, goTo, currentPage } = usePagination(encounters ?? [], encountersCount);
+  const isTablet = useLayoutType() === 'tablet';
+  const [filteredRows, setFilteredRows] = useState<Array<MappedEncounter>>([]);
+  const [filter, setFilter] = useState('');
+
+  useEffect(() => {
+    if (filter) {
+      setFilteredRows(encounters?.filter((encounter) => encounter.encounterType === filter));
+      setFilter('');
+    }
+  }, [filter, filteredRows, encounters]);
 
   const tableHeaders = [
     {
       id: 1,
-      header: isShowingAllEncounters ? 'Date & time' : 'Time',
+      header: showAllEncounters ? 'Date & time' : 'Time',
```

Comment:
```suggestion
      header: showAllEncounters ? t('dateTime','Date & time') : t('time','Time'),
```
We should have these table headers translated